### ISSUE TITLE
balance doesn't change

### ISSUE THREAD

**Issue by useroop** [iron-fish/ironfish#2551]
### What happened?

During few transactions IRON balance didn't change. 
These are sequential transactions [diff time ~ 1 sec] from my side:
e70a79715a1b1d6134a89fbffe649fdc3c468f389284366ab688be1cdb48af72
f820503cb78961ceda8ebbb8e54c79d3f6176502086fcae91c58c018ad22c00c
0eccae2e49469a03fb98dba1ea6a319b145f31bbd25a7bc2ecfade0247420fcd

![image](https://user-images.githubusercontent.com/43512615/201235773-53bf184d-099c-4e8f-90ba-a35c90c41bbc.png)

### Version

0.1.51

### Debug output

```shell
Iron Fish version               0.1.51 @ 67da183
Iron Fish library               0.0.28 @ 67da183
Operating system                Linux x64
CPU model(s)                    AMD Ryzen 5 3600 6-Core Processor
CPU threads                     12
RAM total                       62.74 GiB
Heap total                      4.01 GiB
Node version                    v16.17.0
ironfish in PATH                true
Garbage Collector Exposed       false
Telemetry enabled               true
Node name                       ironfish_1
Block graffiti                  ironfish_1
```

### Graffiti

ironfish_1

**Comment by hughy** [comment]
@useroop do these transaction hashes appear on the block explorer?

**Comment by useroop** [comment]
@hughy no as you can check by yourself as I posted its hashes.

But I also don't see transactions which I sent from one account to another despite the balance was changed. So I don't believe explorer for 100%.

**Comment by hughy** [comment]
You posted the hashes in an image, and it's very difficult for me to copy them out by hand because each hash is 64 characters long.

You can use the `ironfish accounts:transactions` command to see the statuses of your transactions. It's possible that your balance has changed because the transaction you've sent between accounts is `pending`, but does not appear on the block explorer because it is not on the main chain.

**Comment by useroop** [comment]
@hughy I wrote down transactions hashe in the very beginning of my text message. And balance hasn't changed.
I've checked all of them by using additional flag -t --> all are expired. But it isn't an answer on my question cause I asked and showed that the balance hasn't changed in that moment. Even if transaction goes to pending it must take some of balance doesn't it?

**Comment by hughy** [comment]
Sorry, my mistake, I didn't notice the hashes at the beginning of the message.

It's likely that your wallet database has a mismatch between its balance and its unspent notes. We've noticed that notes from transactions created on forks are causing problems with balances and we're currently working on resolving bugs related to the way the wallet handles transactions created on forks.

You may need to rescan to resolve the balance discrepancy. Setting `minimumBlockConfirmations` to a higher value (default is 2) can also help to avoid creating transactions on forks.

**Comment by hughy** [comment]
Are you still seeing this issue?

Could you share the result of `config:get defaultTransactionExpirationSequenceDelta`?

**Comment by useroop** [comment]
The result is 15.

Now I don't have an IRON balance so cannot face with this issue.

**Comment by hughy** [comment]
Were you able to deposit your balance?

Did you run `accounts:rescan`?

**Comment by useroop** [comment]
My shown balance is 0. If I use rescan I'll lose few days so I'm not sure that it'll make sense.

What do you think about it?

**Comment by hughy** [comment]
I asked because I wondered how your balance got to 0.

If you expect your balance to be more than 0 then it may make sense to rescan

**Comment by useroop** [comment]
Not all transactions were unsuccessful.

I don't understand how it should work (I mean command -deposit) cause some times after sending transaction my balance decreased by 20 IRONs and than almost 20 IRONs came back to balance. With each new transaction the IRON amount withrawal from the balance decreased (but than came back). Than I faced with a sharp decrease in my balance (about 20 IRONs). For some period of time I have been facing the described issue which I have reported when the balance didn't change.

So I cannot calculate and understand if I should have left something on my balance.

**Comment by hughy** [comment]
Transactions in UTXO-based cryptocurrencies like Iron Fish (and bitcoin) work a bit like this:

- Your balance is made of 'notes' or 'coins', which are unique and each have a specific value.
- When you make a transaction, you spend the notes in your account. So if you have one note worth $IRON 20, but you only want to send $IRON 0.1, then you have to spend your note worth 20 and you get back another note worth 19.9 as 'change'

- If your transaction goes through and is confirmed on the chain, then your balance will be 19.9
- If it expires, then you keep your original note and your balance goes back to 20


If you run `ironfish accounts:balance --explain` you will see three values: the 'confirmed' balance which is notes that you can spend, the 'unconfirmed' balance which is notes that are on the chain but too recent to spend, and 'pending' which is the change from your transactions that haven't gone through yet.

What is the result of that command for you?

**Comment by hughy** [comment]
The transaction may appear as pending if your accounts database is not synced up to the chain yet.

If you run `accounts:status`, the `Head Sequence` shows where each account is synced up to

**Comment by useroop** [comment]
Account Name        Account ID          Head Hash Head In Chain Head Sequence
 ─────────────────── ─────────────────── ───────── ───────────── ─────────────
 default             50a407eb-147a-4076… 00000000… true          273961
 oop_wallet_ironfish c3583aa6-a47b-45d4… 00000000… true          273961

**Comment by hughy** [comment]
> I had 40 IRONs and than received 10 additional IRONs -- so now I should have 400 transactions (40 IRONs / 0.1 IRON) + 10IRONs (from pending), but the fact is 80 transactions + 0 balance (maybe 10 IRON will come back later).

We recently increased the amount of IRON in deposits from 0.1 to 0.5, so that may explain why you're seeing fewer transactions.

Do you see any transactions from `ironfish accounts:transactions --filter status=pending`?

It looks like you have one transaction that may be "stuck" with its notes in a pending state, but i think it may actually be expired.

**Comment by useroop** [comment]
Nothing here

 Status     Creator Hash Miner Fee Fee ($IRON)         Notes Spends Expiration Sequence
 ────────── ─────── ──── ───────── ─────────────────── ───── ────── ───────────────────

**Comment by hughy** [comment]
If you run `accounts:notes`, is there a note with a value of `9.99900035`?

If so, can you run `ironfish accounts:transaction <hash>` for its transaction hash?

**Comment by useroop** [comment]
![image](https://user-images.githubusercontent.com/43512615/202535602-ba82e3e6-0f37-4184-9b70-427747adc55e.png)

**Comment by useroop** [comment]
Also interesting (have 2 such transactions)
![image](https://user-images.githubusercontent.com/43512615/202535986-dc432e39-69e3-427c-9b3d-d285903a8a32.png)

**Comment by hughy** [comment]
Thanks again, this is a huge help!

It looks like the transaction has expired, but the note is stuck being counted as pending. I'll see what I can do to reproduce this

**Comment by hughy** [comment]
The negative fee value is for miner rewards -- your account was paid the fee instead of the other way around!

**Comment by hughy** [comment]
There's a fix for this issue in #2595 , which we hope to get out soon!

**Comment by hughy** [comment]
Resolved with resolution to #2622
